src/store/undoRedo/index.ts

```diff
@@ -116,14 +120,32 @@ export const mutations: MutationTree<IUndoRedoState> = {
 
 export const actions: ActionTree<IUndoRedoState, IRootState> = {
   async handleStateChange({commit, getters, rootState}) {
-    // Take a snapshot of the current state to avoid mutating> it
+    const currentModules: ISnapshotModules = structuredClone({flows: getters.currentSnapshot?.modules.flows ?? {}})
+    const newModules: ISnapshotModules = structuredClone({flows: rootState.flow})
+    const changedKeys = getChangedKeys(currentModules, newModules)
```

Comment:
@object-required
I had to move the `getChangedKeys` call to the top, so that it could be used for deep link generation.
But there is no `newSnapshot` available here (it needs the `routeName` & `routeParams` to be complete - vicious circle).
So I used `structuredClone` on modules directly instead of `pack` function.

@safydy-r suggested we use `pack` (and I agree). If you think this is worth doing and you know how - please suggest! :)

Reply:
When thinking, I'm ok with this implementation as newSnapshot is not available yet

Reply:
I'm ok as well, we can refactor later if/when it bothers us :wink: